src/SelfSignedCertificate.cpp

```diff
@@ -39,7 +39,6 @@ bool SelfSignedCertificate::generate(QString clientCertName, QString clientCertE
 	ASN1_TIME *notBefore = NULL;
 	ASN1_TIME *notAfter = NULL;
 	QString commonName;
-	bool isServerCert = clientCertName.isEmpty() && clientCertEmail.isEmpty();
```

Comment:
Let's keep this to keep the code simple.
Just let `bool isServerCert = certificateType == CertificateTypeServerCertificate;` or so.

Reply:
... and drop most other changes in this function.